meson.build

```diff
@@ -0,0 +1,75 @@
+# Meson build script from MSFITSLIB
+project('msfitslib', 'cpp', version: '1.0.1',
+         default_options : ['cpp_std=c++11',])
+
+# Basic setup
+cc  = meson.get_compiler('cpp')
+builddir = meson.project_build_root()
+
+comp_flags = ['-D_UNIX','-D_ENABLE_PROFILER_', '-D_ENABLE_TRACING_', 
+              '-D_ENABLE_VISUALIZATION_', '-D_MONTE_CARLO_', '-D_NO_ROOT_']
+add_project_arguments(comp_flags, language : 'cpp')
+
+# Load list of files to compile into shared object
+# Note: Meson does not allow * wildcard (deliberately)
+# Instead, we use file generated from command: ls src/*.cpp > meson_srclist.txt
+fs = import('fs') 
+srcs = fs.read('meson_srclist.txt').strip().split('\n')
+
+# Dependencies: FFTW3, CFITSIO, LIBNOVA, LDL, PTHREAD
+fftw3_dep = dependency('fftw3')
+cfitsio_dep = dependency('cfitsio')
+libnova_dep = cc.find_library('libnova', dirs : '/usr/lib/x86_64-linux-gnu') 
+ldl_dep     = cc.find_library('dl') 
+lpthread_dep = cc.find_library('pthread')
+
+# Compile shared object
+msfitslib = library('msfitslib', srcs,              
+                     include_directories: 'src',
+                     dependencies: [fftw3_dep, cfitsio_dep, libnova_dep, ldl_dep, lpthread_dep],
+                     install: true
+                    )
+
+# Install headers
+install_headers('src/array_config_common.h', 'src/basestring.h', 'src/cvalue_vector.h',
+                'src/libnova_interface.h',
+                'src/bg_fits.h', 'src/bg_array.h','src/bg_globals.h', 'src/bg_date.h', 
+                'src/bg_defines.h', 'src/bg_total_power.h', 
+                'src/mystring.h', 'src/myfile.h', 'src/mytypes.h', 'src/basedefines.h',
+                'src/mystrtable.h', 'src/mylock.h', 'src/mypipe.h', 'src/mydate.h')
```

Comment:
Just as an aside, "mytypes.h" etc. are pretty vague include names, and installing them directly to includedir can lead to namespace issues if third-party consumers are supposed to use `#include <mytypes.h>`.

However, this is equal to what cmake does now...

Reply:
I thought this was non-optimal as I was doing it 😱. 

Perhaps there's a better approach: Is there another way to share the location of header files in `subproject_A` with `subproject_B`? 

As context, these headers are needed by another codebase (which I've also added a `meson.build` script to). I was planning to build both codes as meson subprojects.

Reply:
There's a couple common tactics in the C/C++ ecosystem for organizing headers. Assume we use "msfits" as a project identifier:
- you could install each header to `/usr/include/msfits/*.h` (add `subdir: 'msfits'` to `install_headers`) and add `${includedir}/msfits` as an include directory in pkg-config(`subdirs: 'msfits'` in `pkgconfig.generate()`), to hide the headers from any software that doesn't actually depend on the pkg-config file `msfitslib.pc`
- you could move all headers to `<project_root>/include/msfits/*.h`, install them to `/usr/include/msfits/*.h`, and have external consumers use `#include <msfits/*.h>`; add `include/` in Meson for `msfitslib_dep`
- rename each header to e.g. `msfits_basestring.h`

Different projects prefer different styles. All that really matters in the end is that it can get confusing to have a file /usr/include/basestring.h that might belong to any project, and lead to confusing errors when compiling completely unrelated software.

The first option is a pretty simple solution that doesn't require updating source code with new header names. I personally happen to like the second option for "I'm starting a new project, no one is relying on my header names yet, what's the best project skeleton to go with for long-term flexibility and organization".

Reply:
And again: this isn't really a problem with this meson.build, it's just a general concern that was already there in the cmake files too, which I happened to notice.